waspc/src/Wasp/Generator/WebAppGenerator/RouterGenerator.hs

```diff
@@ -59,35 +60,39 @@ instance ToJSON PageTemplateData where
         "importFrom" .= _importFrom pageTD
       ]
 
-generateRouter :: AppSpec -> FileDraft
-generateRouter spec =
-  C.mkTmplFdWithDstAndData
-    (asTmplFile $ [reldir|src|] </> routerPath)
-    targetPath
-    (Just $ toJSON templateData)
+generateRouter :: AppSpec -> Generator FileDraft
+generateRouter spec = do
+  templateData <- createRouterTemplateData spec
+  return $
+    C.mkTmplFdWithDstAndData
+      (asTmplFile $ [reldir|src|] </> routerPath)
+      targetPath
+      (Just $ toJSON templateData)
   where
     routerPath = [relfile|router.js|]
-    templateData = createRouterTemplateData spec
     targetPath = C.webAppSrcDirInWebAppRootDir </> asWebAppSrcFile routerPath
 
-createRouterTemplateData :: AppSpec -> RouterTemplateData
-createRouterTemplateData spec =
-  RouterTemplateData
-    { _routes = routes,
-      _pagesToImport = pages,
-      _isAuthEnabled = AS.isAuthEnabled spec
-    }
+createRouterTemplateData :: AppSpec -> Generator RouterTemplateData
```

Comment:
Same as for the function below, I think we can remove Generator since it is a helper function and not generator function? If we figure out in the future we need to make it Generator, we can always add it back, but right now it seems redundant?

Reply:
Yep good call, I may have gone overboard on those two! :D thanks